### ISSUE TITLE
ResNet output from CSEvalClientTest stays the same regardless of input

### ISSUE THREAD

**Issue by achhibbar** [Microsoft/CNTK#1017]
I am using a ResNet_18 model I have trained and tested that classifies into two categories. Using CNTK.exe it seems to be producing good results on my test data.
However when I test this model with CSEvalClientTest, the output never changes regardless of what input image I use.
Ouput[0] always equals 0.150724843
Ouput[1] always equals 0.150217175
Alexnet is working, with outputs varying depending on the input image.

**Comment by zhouwangzw** [comment]
@achhibbar Could you please check whether the inputs to the model.Evaluate(), i.e. after image pre-processing, are the same for different input images? CNTK.exe and CSEvalClientTest use different ways for image pre-processing. I tried the EvaluateImageClassificationModel() in CSEvalClient with different images and got different results.

**Comment by achhibbar** [comment]
I confirmed that the inputs are different for each image.
I also confirmed that these inputs work fine with the ResNet_18.model from https://www.cntk.ai.
They also work with my trained AlexNet model.
Could this have to do with the fact that my ResNet model used Batch Normalization? The pre-trained ResNet model does not use BN (as you mentioned previously), and neither does the AlexNet model.

**Comment by achhibbar** [comment]
I changed 
`y = BatchNormalization(inp, sc, b, m, v, spatial = true, normalizationTimeConstant = bnTimeConst, epsilon = 0.000000001, imageLayout = "cudnn", engine = "cudnn")`
to
`y = BatchNormalization(inp, sc, b, m, v, spatial = true, normalizationTimeConstant = bnTimeConst, epsilon = 0.000000001, useCntkEngine = true)`
Testing an early trained model suggests this may have worked - I will know more tomorrow.

**Comment by achhibbar** [comment]
Now I am seeing my outputs vary as my input images changes, however the outputs don't have the values I expect.
For ResNet_18, the 2 outputs are identical to two decimal places, except one is negative and the other is positive. Example outputs:
```
4.26857 , -4.265889
-0.4408314 , 0.4410337
```
I do not see this with my AlexNet model; the two outputs are always different.
Could this have something to do with BN?

**Comment by zhouwangzw** [comment]
@achhibbar Do you get the same outputs when using "eval" command with cntk.exe (using GPU)?

**Comment by achhibbar** [comment]
Can you point me to some examples that show how to look at the outputs when using the "eval" command with cntk.exe? I have used cntk.exe to test/eval the error across a large set of images, but not to assess the output values for a single image.
